File: app/src/main/java/com/harrywoodworth/atoll/game/island/ForestGrowth.java
package com.harrywoodworth.atoll.game.island;

import com.harrywoodworth.atoll.game.utility.DesignManager;

public class ForestGrowth {

    // How efficient is the Forest at growing? Determines chance for forest spread
    public static double forest_evolution_rate = DesignManager.DEFAULT_FOREST_EVOLUTION_RATE;

    // How long has the forest been growing? Determines number of loops of forest growth
    public static int growth_factor = DesignManager.DEFAULT_FOREST_GROWTH_FACTOR;

    public ForestGrowth(){}

    public ForestGrowth(double _forest_evolution_rate, int _growth_factor) {
        if(_forest_evolution_rate != 0.0)
            forest_evolution_rate = _forest_evolution_rate;
        if(_growth_factor != 0)
            growth_factor = _growth_factor;
    }



}

File: app/src/main/java/com/harrywoodworth/atoll/game/island/ForestGrowth.kt
package com.harrywoodworth.atoll.game.island

import com.harrywoodworth.atoll.game.utility.DesignManager

/// Evolution rate is how many times it grows per factor, factor is how many times it evolves
data class ForestGrowth (var forest_evolution_rate: Double = DesignManager.DEFAULT_FOREST_EVOLUTION_RATE,
                         var growth_factor: Int = DesignManager.DEFAULT_FOREST_GROWTH_FACTOR,
                         var seed_count : Int = DesignManager.DEFAUL_FOREST_SEED_COUNT)